src/framework/theme/components/chat/chat-custom-message.directive.ts

```diff
@@ -0,0 +1,45 @@
+import { Directive, Input, OnInit, TemplateRef } from '@angular/core';
+
+@Directive({
+  selector: `[nbCustomMessage]`,
+})
+export class NbChatCustomMessageDirective implements OnInit {
+
+  @Input() set nbCustomMessage(content: string) {
+    if (!content) {
+      throwCustomMessageContentIsMissed();
+    }
+    this._type = content;
+  }
+
+  get type(): string {
+    return this._type
+  }
+  protected _type: string;
+
+  constructor(public templateRef: TemplateRef<any>) {
+    if (!this.templateRef) {
```

Comment:
We could remove `templateRef` check as if there is no template, Angular throws before constructor execution.